plugin/keyring_vault/vault_curl.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MYSQL_VAULT_CURL_H
+#define MYSQL_VAULT_CURL_H
+
+#include <my_global.h>
+#include <curl/curl.h>
+#include "i_vault_curl.h"
+#include "logger.h"
+#include "vault_credentials.h"
+#include "i_keyring_key.h"
+
+namespace keyring
+{
+
+class Vault_curl : public IVault_curl
+{
+public:
+  Vault_curl(ILogger *logger)
+    : logger(logger)
+    , curl(NULL)	
+    , list(NULL)
+  {}
+
+  ~Vault_curl()
+  {
+    if (list != NULL)
+      curl_slist_free_all(list);
+    curl_easy_cleanup(curl);
+  }
+
+  my_bool init(Vault_credentials *vault_credentials);
```

Comment:
Why do we need the 'init()' function?
can we do initialization in the constructor?